### PR TITLE
perf: only upsert once when fetching asset market data

### PR DESCRIPTION
**Pull request by woodenfurniture**
## Description

Improves local app performance by minimizing the number of redux state updates during app boot.
## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [x] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

NA

## Risk

Moderate risk change affecting how asset prices are loaded during app boot.

## Testing

Clear cache and check that asset prices are loaded correctly.

### Engineering

☝️

### Operations

☝️

## Screenshots (if applicable)

The following shows performance improvement locally. Please note that the performance improvement in production will be far less noticeable.

Develop (local) - 1.43s:
image

This PR (local) - 0.04s:
image

### CODE REVIEW

**Comment by woodenfurniture** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #5327** <a href="https://app.graphite.dev/github/pr/shapeshift/web/5327" target="_blank">Graphite</a> 
    * **PR #5328** <a href="https://app.graphite.dev/github/pr/shapeshift/web/5328" target="_blank">Graphite</a> 
      * **PR #5329** <a href="https://app.graphite.dev/github/pr/shapeshift/web/5329" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/5329?utm_source=stack-comment).

**Review by 0xApotheosis — APPROVED** [review]

File: src/context/AppProvider/AppContext.tsx [hunk 1]
```diff
@@ -273,15 +274,20 @@ export const AppProvider = ({ children }: { children: React.ReactNode }) => {
...
+          marketApi.endpoints.findPriceHistoryByAssetIds.initiate({
+            timeframe,
+            assetIds: portfolioAssetIdsExcludeNoMarketData,
+          }),
+        ),
+        dispatch(marketApi.endpoints.findByAssetIds.initiate(portfolioAssetIdsExcludeNoMarketData)),
+      ])
+
+      // used to trigger mixpanel init after load of market data
+      dispatch(marketData.actions.setMarketDataLoaded())
```
Comment (gomesalexandre) [inline src/context/AppProvider/AppContext.tsx:289]: 🧠

File: src/state/slices/marketDataSlice/marketDataSlice.tsx [hunk 2]
```diff
@@ -153,22 +157,38 @@ export const marketApi = createApi({
...
+      queryFn: async function findByAssetIds(assetIds: AssetId[], { dispatch }) {
+        if (assetIds.length === 0) return { data: null }
+
+        const responseData = await Promise.all(
+          assetIds.map(async assetId => {
+            try {
+              const currentMarketData = await getMarketServiceManager().findByAssetId({ assetId })
+              return { assetId, currentMarketData }
+            } catch (e) {
+              return { assetId, currentMarketData: null }
```
Comment (gomesalexandre) [inline src/state/slices/marketDataSlice/marketDataSlice.tsx:171]: Ditto logging here?

**Review by gomesalexandre — APPROVED** [review]
Tested with the same testing criteria as https://github.com/shapeshift/web/pull/5328, same non-blocking comment, and found no regressions here as well 🎉

**Pull request merged**